src/test/groovy/org/mockito/release/gradle/ReleaseNeededTaskTest.groovy

```diff
@@ -4,12 +4,47 @@ import org.gradle.api.GradleException
 import org.gradle.testfixtures.ProjectBuilder
 import org.mockito.release.internal.comparison.PublicationsComparatorTask
 import spock.lang.Specification
+import spock.lang.Unroll
 
 class ReleaseNeededTaskTest extends Specification {
 
     def tasks = new ProjectBuilder().build().getTasks();
     ReleaseNeededTask underTest = tasks.create("releaseNeeded", ReleaseNeededTask)
 
+    @Unroll
+    def "release is needed" (commitMessage, branch, pullRequest, publicationsComparatorsResults, releaseNotNeeded){
```

Comment:
This is a very good test. Normally I'm against using imperative loops inside test method but in this case it makes sense to get good coverage. It covers the conditional logic that is likely to break when we change stuff. Thank you!